include/circt/Dialect/FIRRTL/InstanceGraph.h

```diff
@@ -113,6 +116,10 @@ class InstanceGraphNode {
   /// List of instances which instantiate this module.
   UseVec moduleUses;
 
+  ///  The depth of a node in a DAG is the length of the longest path from a
+  ///  source to the node.
+  size_t depth;
```

Comment:
One concern with keeping depth *in* hte InstanceGraph (as opposed to in an on-the-side map), is that this means that all clients will have to keep this up to date.  If we have inlining or other things that mutate the instance graph and update the InstanceGraph data structure on the fly, this could be a pain.  @youngar what do you think about this?

Reply:
I think it makes sense to store pre-processing data in a side table that we can memoize.   I haven't read through the paper that @seldridge linked, but maybe that one is easier to incrementally update?

Reply:
There's a sparse table in there, so it would need to be reconstructed on any change to the hierarchy. 😢 I don't think incremental updates are possible for that formulation. (I don't know the space of LCA algorithms for mutable trees. There's a prior in the SFC LCA implementation to do lots of queries on a fixed hierarchy because the Wiring Transform is batch resolving all the source/sink annotations.)